### PR TITLE
chore: set max SDK version to <3.0.0

### PR DESCRIPTION
**Pull request by chalin**
cc @kwalrath @kevmoo

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,2 +1 @@
 analyzer:
```
Comment (nex3) [inline analysis_options.yaml]: Just delete this file entirely if it's not configuring anything.

File: pubspec.yaml [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+  
 dependencies:
-  charcode: "^1.1.0"
-  collection: ">=1.1.0 <2.0.0"
-  string_scanner: ">=0.1.4 <2.0.0"
-  source_span: ">=1.0.0 <2.0.0"
+  charcode: ^1.1.0
+  collection: ^1.14.10
+  string_scanner: ^1.0.2
+  source_span: ^1.4.0
```
Comment (nex3) [inline pubspec.yaml]: Why are you narrowing these constraints?

File: test/utils.dart [hunk 3]
```diff
@@ -7,7 +7,7 @@ import 'package:yaml/src/equality.dart' as equality;
 import 'package:yaml/yaml.dart';
 
 /// A matcher that validates that a closure or Future throws a [YamlException].
-final Matcher throwsYamlException = throwsA(new isInstanceOf<YamlException>());
+final Matcher throwsYamlException = throwsA(TypeMatcher<YamlException>());
```
Comment (nex3) [inline test/utils.dart]: I believe there are still frontends in use, at least within Google, that don't yet support optional `new`.

Also, if you are using it, you should update the SDK constraint to exclude any SDKs where frontends don't use Dart 2 by default.

File: analysis_options.yaml (hunk 1 above)
Comment (chalin) [reply analysis_options.yaml]: done

File: pubspec.yaml (hunk 2 above)
Comment (chalin) [reply pubspec.yaml]: (For the reasons I explained yesterday, but I'll add them back)

File: test/utils.dart (hunk 3 above)
Comment (chalin) [reply test/utils.dart]: Adding `new` back

File: pubspec.yaml [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
+  charcode: ^1.1.0
+  collection: '>=1.1.0 <2.0.0'
+  string_scanner: '>=0.1.4 <2.0.0'
+  source_span: '>=1.0.0 <2.0.0'
+
 dev_dependencies:
-  path: ">=1.2.0 <2.0.0"
-  test: ">=0.12.0 <0.13.0"
+  path: ^1.6.1
+  test: ^1.2.0
```
Comment (nex3) [inline pubspec.yaml]: These dependencies also shouldn't be narrowed.

Comment (chalin) [reply pubspec.yaml]: Oops, sorry for having missed those. Done

**Pull request merged**